starlette/websockets.py

```diff
@@ -138,13 +138,20 @@ async def send_json(self, data: typing.Any, mode: str = "text") -> None:
         else:
             await self.send({"type": "websocket.send", "bytes": text.encode("utf-8")})
 
-    async def close(self, code: int = 1000) -> None:
-        await self.send({"type": "websocket.close", "code": code})
+    async def close(self, code: int = 1000, reason: typing.Optional[str] = "") -> None:
+        if reason != "":
+            await self.send({"type": "websocket.close", "code": code})
+        else:
+            await self.send({"type": "websocket.close", "code": code, "reason": reason})
```

Comment:
```suggestion
    async def close(self, code: int = 1000, reason: str = None) -> None:
        if reason is None:
            await self.send({"type": "websocket.close", "code": code})
        else:
            await self.send({"type": "websocket.close", "code": code, "reason": reason})
```

Reply:
This seems like the proper way. I might add that we would need to change reasons type (`str`) to be `typing.Optional[str]` but this would be implicated anyway given the default argument.